ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/manager/task/RetryingGetBlockFromPeersTaskTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright contributors to Hyperledger Besu
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth.manager.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hyperledger.besu.ethereum.core.BlockHeader.GENESIS_BLOCK_NUMBER;
+import static org.mockito.Mockito.mock;
+
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.eth.manager.EthPeer;
+import org.hyperledger.besu.ethereum.eth.manager.ethtaskutils.RetryingSwitchingPeerMessageTaskTest;
+import org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerTask.PeerTaskResult;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class RetryingGetBlockFromPeersTaskTest
+    extends RetryingSwitchingPeerMessageTaskTest<PeerTaskResult<Block>> {
+
+  @Override
+  protected void assertResultMatchesExpectation(
+      final PeerTaskResult<Block> requestedData,
+      final PeerTaskResult<Block> response,
+      final EthPeer respondingPeer) {
+    assertThat(response.getResult()).isEqualTo(requestedData.getResult());
+  }
+
+  @Override
+  protected PeerTaskResult<Block> generateDataToBeRequested() {
+    final Block block = blockchain.getBlockByNumber(10).get();
+    return new PeerTaskResult<>(mock(EthPeer.class), block);
+  }
+
+  @Override
+  protected RetryingGetBlockFromPeersTask createTask(final PeerTaskResult<Block> requestedData) {
+    return RetryingGetBlockFromPeersTask.create(
+        protocolSchedule,
+        ethContext,
+        metricsSystem,
+        maxRetries,
+        Optional.of(requestedData.getResult().getHash()),
+        GENESIS_BLOCK_NUMBER);
+  }
+
+  @Test
+  @Override
+  @Ignore("GetBlock could not return partial response")
+  public void failsWhenPeerReturnsPartialResultThenStops() {}
```

Comment:
is this new? do we plan on fixing these tests?

Reply:
this is part of the parent class `RetryingMessageTaskTest` , that has tests for use cases that do not always apply to actual derived test classes, in our case we test getting a block from peer, and there is no partial response, either the block is dowloaded or not, so this test is ignored since do no apply. You can find similar ignore in all the other concrete implementation of these abstract test classes.